File: 卡码/代码随想录/1/06-二叉树/Go/04-二叉搜索树属性/LeetCode_0530.go
// 530. 二叉搜索树的最小绝对差
package main

import (
	. "binaryTree/types"
	"math"
)

func getMinimumDifference(root *TreeNode) int {
	var res=math.MaxInt
	var pre *TreeNode=nil
	var traversal func(root *TreeNode)
	traversal=func(node *TreeNode){
		if node==nil {
			return
		}
		traversal(node.Left)
		if pre!=nil {
			res=int(math.Min(float64(res),float64(node.Val-pre.Val)))
		}
		pre=node
		traversal(node.Right)
	}
	traversal(root)
	return res
}

File: 卡码/代码随想录/1/06-二叉树/Js/04-二叉搜索树属性/LeetCode_0530.js
//530. 二叉搜索树的最小绝对差
import TreeNode from "../TreeNode.js"

const getMinimumDifference = function(root) {
  let res=Infinity;
  let pre=null;
  const traversal=node=>{
    if(!node) return;
    traversal(node.left);
    if(pre) res=Math.min(res,node.val-pre.val);
    pre=node;
    traversal(node.right);
  };
  traversal(root);
  return res;
};